src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,93 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+/**
+ * Storage helper for storage related tasks.
+ */
+public class Storage {
+
+    private final String filepath;
+    private final ArrayList<Task> list = new ArrayList<>();
+
+
+    /**
+     * Constructor for Storage class.
+     *
+     * @param filepath path for saving and loading
+     */
+    public Storage(String filepath) {
+        this.filepath = filepath;
+    }
+
+    /**
+     * Return list of task loaded from save file.
+     *
+     * @return List of Tasks
+     * @throws DukeException Exceptions for file not found
+     */
+    public ArrayList<Task> load() throws DukeException {
+        File file = new File(filepath);
+        try {
+            Scanner scanner = new Scanner(file);
+            while (scanner.hasNextLine()) {
+                String data = scanner.nextLine();
+                Task t = processData(data);
+                if (t != null) {
+                    list.add(t);
+                }
+            }
+            scanner.close();
+        } catch (FileNotFoundException e) {
+            throw new DukeException("File is not found");
+        }
+        return list;
+    }
+
+    private Task processData(String str) throws DukeException {
+        Scanner stringProcessor = new Scanner(str);
+        stringProcessor.useDelimiter("\\|");
+        try {
+            String type = stringProcessor.next();
+            switch (type) {
+            case "T":
+                return new ToDo(stringProcessor.next(), stringProcessor.next());
+            case "D":
+                return new Deadline(stringProcessor.next(), stringProcessor.next(), stringProcessor.next());
+            case "E":
+                return new Event(stringProcessor.next(), stringProcessor.next(), stringProcessor.next());
+            }
+            return null;
+        } catch (NoSuchElementException e) {
+            throw new DukeException("Save file has invalid format.");
+        }
+    }
+
+    void saveFile(ArrayList<Task> list) throws DukeException {
```

Comment:
Is the method access modifier deliberately omitted here?

Reply:
Perhaps you can add a `public` access modifier to this method to make it clear that it is publicly accessible?